compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs

```diff
@@ -286,6 +286,11 @@ impl<'tcx> GotocCtx<'tcx> {
             "atomic_xadd_acqrel" => codegen_atomic_binop!(plus),
             "atomic_xadd_rel" => codegen_atomic_binop!(plus),
             "atomic_xadd_relaxed" => codegen_atomic_binop!(plus),
+            "atomic_xchg"
+            | "atomic_xchg_acq"
+            | "atomic_xchg_acqrel"
+            | "atomic_xchg_rel"
+            | "atomic_xchg_relaxed" => self.codegen_atomic_store(intrinsic, fargs, p, loc),
```

Comment:
I would split into several arms like we did for all other cases.

Reply:
done